integration-tests/tests/src/tests/dictionary.ts

```diff
@@ -333,4 +441,262 @@ describe("Dictionary", () => {
     },
     expectedError: "JS value must be of type 'object'",
   });
+  describe("Dictionary queries", () => {
+    openRealmBeforeEach({ schema: [DictSchema] });
+    it("support filters", function (this: RealmContext) {
+      const N = 100;
+      for (let i = 0; i < N; i++) {
+        this.realm.write(() => this.realm.create(DictSchema.name, { fields: { x: i, y: 2, z: 3 } }));
+      }
+
+      const data = this.realm.objects(DictSchema.name);
+      expect(data.length).equals(N, "We expect ${N} objects.");
+
+      const half = data.filtered("fields['x'] >= 50");
+      const seventy = data.filtered("fields['x'] >= $0", 70);
+      expect(half.length).equals(N / 2, "We expect only 50 items, matching for field x.");
+      expect(seventy.length).equals(30, "We expect only 30 items, matching for field x.");
+    });
+  });
+  describe("notifications", () => {
+    openRealmBeforeEach({ schema: [DictSchema] });
+    it("support update notifications", async function (this: RealmContext) {
+      const UPDATES = 5;
+      this.realm.write(() => this.realm.create(DictSchema.name, { fields: { field1: 0, filed2: 2, field3: 3 } }));
+      const fields = this.realm.objects<IDictSchema>(DictSchema.name)[0].fields;
+      let cnt = 0;
+      fields.addListener((_: any, changeset: { modifications: any[] }) => {
+        expect(fields.field1).equals(cnt, "fields.field1: ${fields.field1} should be equals to: cnt -> ${cnt}");
+        // We ignore the first as it just reflect the creation in the line above.
+        if (cnt > 0) {
+          expect(changeset.modifications[0]).equals(
+            "field1",
+            "The changeset should reflect an update on field1 but it shows -> ${changeset.modifications[0]}",
+          );
+        }
+        cnt++;
+      });
+
+      for (let i = 1; i <= UPDATES; i++) {
+        this.realm.write(() => {
+          fields.field1 = i;
+        });
+      }
+
+      await sleep(1000);
+      expect(this.realm.objects<IDictSchema>(DictSchema.name)[0].fields.field1).equals(5);
+      expect(cnt).equals(UPDATES + 1, "We expect ${UPDATES + 1} updates.");
+      fields.removeAllListeners();
+      this.realm.close();
+    });
+    // FIXME: Running the following test produce unexpected side-effects which breaks all subsequent tests.
+    // See https://github.com/realm/realm-js/issues/3834
+    it.skip("support insert notifications", async function (this: RealmContext) {
```

Comment:
Did you verify this is still the case?

Reply:
thought I did but after having a second glance at it I realised that I had written the tests in an invalid way, should work now! (If CI lets it through)